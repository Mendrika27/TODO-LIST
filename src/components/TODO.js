import "./../App.css";
import React, { useState } from "react";
import Header from "./Header";
export default function(){
    const [state,setState]=useState([])
    const [done,setdone]=useState([])
    return(
        <div className="ok">
            <Header/>
            <div className="row">
                <Todo state={state} setState={setState} done={done} setdone={setdone}/>
                <Done done={done} />
            </div>
            
        </div>
    );
}

function Todo({state, setState, done, setdone}){
    
    const [input,setInput]=useState("")
    const inputValue = (str)=>{
        setInput(str)
    }
    const save = (str)=>{
        setState((e)=>[...e,str]);
    }
    const remove = (e,index)=>{
        e.preventDefault();
        const someState=state;
        setdone([...done,state[index]])
        setState(done)
        someState.splice(index,1)
        setState([...someState])

    }
    return(
        <div className="body">
            <div className="input">
                <input type={"text"} onChange={(onChangeValue)=>{inputValue(onChangeValue.target.value)}} onKeyDown={(e)=> e.key === "Enter" && save(input)}></input>
              
            </div>
            
            <div className="todo">
                <h2>To Do</h2>
                {state.map((data,id)=>(<li key={id}><input type={"checkbox"} onClick={(e)=>{remove(e,id)}}></input>{data}</li>))}
            </div>
        </div>
        
    )
}

function Done({done}){
    return(
        <div className="done">
                <h2>Done</h2>
            {done.map((value, id)=>(<li key={id}>{value}</li>))}
        </div>
    )
}