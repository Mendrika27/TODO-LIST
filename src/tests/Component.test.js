import {expect, jest, test} from '@jest/globals';
import { render, fireEvent, screen } from "@testing-library/react";
import React from "react";
import Renderer from "react-test-renderer";
import { Done } from "../components/TODO";
import { Todo } from "../components/TODO";

test("create snapshoot",()=>{
    const done = ["value1","value2"]
    const component = Renderer.create(
        <Done done={done} />
    );
    const theComponent = component.toJSON();
    expect(theComponent).toMatchSnapshot();
})
test("create snapshoot",()=>{
    const state = []
    const setState = jest.fn()
    const done = []
    const setdone = jest.fn();
    const component =Renderer.create(
        <Todo state={state} setState={setState} done={done} setdone={setdone}/>
    );
    const theComponent = component.toJSON();
    expect(theComponent).toMatchSnapshot();
})
test("inputValue function updates the state", () => {  
    const setInput = jest.fn();
    const inputValue = (event) => {
      setInput(event.target.value);
    };
    Renderer.create(<input data-testId="custom-element" onChange={inputValue} />); 
    const element =getByTestId('custom-element');
    fireEvent.change(element, { target: { value: ' custom-element'} });
    expect(setInput).toBeInTheDocument();
});
test("should remove a todo item from the list", () => {
    const { getByPlaceholderText, getByText } = render(<Todo />);
    const input = getByPlaceholderText("Add a todo item");
    fireEvent.change(input, { target: { value: "item 1" } });
    fireEvent.keyDown(input, { key: "Enter", code: 13 });
    const item = getByText("item 1");
    fireEvent.click(item.previousSibling);
    expect(item).not.toBeInTheDocument();
});

