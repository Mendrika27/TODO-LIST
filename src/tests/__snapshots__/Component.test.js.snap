// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`create snapshoot 1`] = `
<div
  className="done"
>
  <h2>
    Done
  </h2>
  <li>
    value1
  </li>
  <li>
    value2
  </li>
</div>
`;

exports[`create snapshoot 2`] = `
<div
  className="body"
>
  <div
    className="input"
  >
    <input
      onChange={[Function]}
      onKeyDown={[Function]}
      type="text"
    />
  </div>
  <div
    className="todo"
  >
    <h2>
      To Do
    </h2>
  </div>
</div>
`;
